code/modules/clothing/glasses/hud.dm

```diff
@@ -203,4 +203,150 @@ obj/item/clothing/glasses/hud/security/supergars
 					if("Released")		holder.icon_state = "hudreleased"
 					else
 						continue
-				C.images += holder
\ No newline at end of file
+				C.images += holder
+
+/obj/item/clothing/glasses/hud/atmos
+	name = "Atmospheric Scanner HUD"
+	desc = "An advanced display used to observe atmospheric data in an area."
+	icon_state = "atmoshud"
+	item_state = "atmoshud"
+	action_button_name = "Set HUD Mode"
+	var/intensity = 64 // The value, 0-255, that determines opaqueness of the overlay
+	var/list/images = list()
+	var/list/modes = list("Pressure","Temperature","Off")
+	var/currentmode = 1 // Represents which mode in the list modes is currently active
+	var/const/basetemp = 293.15
+	var/const/basepressure = 101.325
+
+/obj/item/clothing/glasses/hud/atmos/proc/select_mode()
+	if (!modes.len) return //No possible modes? Must be broked
+	if (currentmode < 1) //Just Checking to prevent problems
+		currentmode = 1
+	else currentmode++
+	if (currentmode > modes.len)
+		currentmode = 1
+
+	for (var/turf/T in images) //After switching modes, erase the images we have.
+		janitor(T)
+	return
+
+/obj/item/clothing/glasses/hud/atmos/attack_self(mob/user)
+	select_mode()
+	usr << "You switch the HUD to [modes[currentmode]]."
+	add_fingerprint(user)
+	..()
+
+/obj/item/clothing/glasses/hud/atmos/equipped(mob/M, slot)
+	if (slot != slot_glasses)
+		for (var/turf/I in src.images)
+			janitor(I)
+	..()
+
+/obj/item/clothing/glasses/hud/atmos/proc/janitor(var/turf/T)

```

Comment:
You want the function name to best describe what it does. janitor() is not very descriptive.